dh.c

```diff
@@ -152,11 +155,48 @@ choose_dh(int min, int wantbits, int max)
 	int best, bestcount, which, linenum;
 	struct dhgroup dhg;
 
+#ifndef WINDOWS
 	if ((f = fopen(_PATH_DH_MODULI, "r")) == NULL) {
 		logit("WARNING: could not open %s (%s), using fixed modulus",
 		    _PATH_DH_MODULI, strerror(errno));
 		return (dh_new_group_fallback(max));
 	}
+#else
+	/* First check the moduli file in the %programdata%\ssh\ directory.
+	 * If not then search for the moduli file in the current executable directory. This file will be updated in new OpenSSH releases.
+	 */
+	if ((f = fopen(_PATH_DH_MODULI, "r")) == NULL) {
+		logit("WARNING: could not open %s (%s)",
```

Comment:
this is the most common scenario and would be logged for every incoming connection. we could remove this warning.

Reply:
fixed it.